chromium_src/chrome/browser/search/instant_service.cc

```diff
@@ -0,0 +1,14 @@
+#include "chrome/browser/search/instant_service.h"
+#include "chrome/browser/search/search_provider_observer.h"
+
+class AlwaysGoogleSearchProviderObserver : public SearchProviderObserver {
```

Comment:
We should check that nothing outside of `InstantService` is having their `SearchProviderObserver` being overridden with this AlwaysGoogle one, because the .cc overriddee has a lot of `#include`s. 
A protection for that _may_ be to have `AlwaysGoogleSearchProvider` _not_ override `SearchProviderObserver` and just a plain class with the `is_google()` member as that's all that `InstantService` uses, although not a guarantee as maybe every use of `SearchProviderObserver` is only to get `is_google()!`
Actually, [looking at Chromium Code Search it seems that the **only** use of `SearchProviderObserver` is `InstantService`](https://source.chromium.org/chromium/chromium/src/+/master:chrome/browser/search/search_provider_observer.h;l=15;bpv=1;bpt=1?q=SearchProviderObserver&ss=chromium%2Fchromium%2Fsrc), so we're ok for now.

Reply:
That's what I noticed too - looks safe to go this route! 👍

Reply:
@petemill did you check the chromium history to see why these features aren't supported? ```// Non-Google NTPs are not supported.``` 
Is it purely a product decision because I'm worried there is an underlying technical issue that could cause problems here

Reply:
I personally believe it's a product decision - opened https://bugs.chromium.org/p/chromium/issues/detail?id=1136372 to hopefully find out more

I dug in with @petemill and basically the use-case for this is for chrome://newtab. If you have Google as your search, it'll show the Google search Doodle, a search box, and then let you do these operations (move the top sites, etc). The observer then detects the search engine change change (one instant service alive for lifetime of app) and forces the "most visited" style of top sites and takes away customization options. If you (for example) set Yahoo as your default, it'll show an empty-ish new tab page with tiles you can't edit and no customize button

Reply:
![image](https://user-images.githubusercontent.com/4733304/95523158-c9bdf500-0982-11eb-9247-537483f4db10.png)